src/app/static/src/js/components/runReport/instances.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+    <div v-if="showInstances">
+        <div v-for="(options, name) in instances" v-if="options.length > 1"
+             id="instances-div" class="form-group row">
+            <label :for="name"
+                   :class="customStyle.label"
+                   class="col-form-label">Database "{{ name }}"</label>
+            <div id="instance-control" :class="customStyle.control">
+                <select class="form-control" :id="name"
+                        @change="handleSelectedInstances"
+                        v-model="selectedInstances[name]">
+                    <option v-for="option in options" :value="option">
+                        {{ option }}
+                    </option>
+                </select>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+    import Vue from "vue"
+    import {ChangelogStyle} from "../../utils/types";
+
+    interface Props {
+        instances: Record<string, any>
+        customStyle: ChangelogStyle
+    }
+
+    interface Data {
+        selectedInstances: Record<string, any>
+    }
+
+    interface Methods {
+        handleSelectedInstances: () => void
+        selectInitialInstance: () => void
+    }
+
+    interface Computed {
+        showInstances: boolean
+    }
+
+    export default Vue.extend<Data, Methods, Computed, Props>({
+        name: "instances",
+        props: {
+            instances: {
+                required: true,
+                type: Object
+            },
+            customStyle: {
+                required: true,
+                type: Object
+            }
+        },
+        data(): Data {
+            return {
+                selectedInstances: {}
+            }
+        },
+        methods: {
+            handleSelectedInstances: function () {
+                this.$emit("selectedValues", this.selectedInstances);
+            },
+            selectInitialInstance: function () {
+                if (this.instances && this.showInstances) {
+                    const instances = this.instances;
+                    for (const key in instances) {
+                        if (instances[key].length > 0) {
+                            this.$set(this.selectedInstances, key, instances[key][0]);
+                        }
+                    }
+                }
+                this.$emit("selectedValues", this.selectedInstances);
+            }
+        },
+        computed: {
+            showInstances() {
+                return !!this.instances;
+            }
+        },
+        mounted() {
+            this.selectInitialInstance();
+        },
+        watch: {
+            instances() {
+                this.selectInitialInstance();
+            },
+            selectedInstances: {
+                deep: true,
+                handler() {
+                    this.$emit("clearRun");
```

Comment:
> This is looking good, a few comments below. Tested with some fake instance metadata and it behaves as expected.
> 
> I think the only unaddressed comment from my last review was for a test of dubious usefulness (updating workflow name when the user can't do that).
> 
> Various comments to do with updating the workflow metadata now covered by [mrc-2511](https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-2511).
> 
> There's a bug associated with the read-only workflow name for Re-run. To reproduce:
> 
> * Re-run existing workflow (goes straight to finalise page)
> * Cancel to go back to start of wizard (OK from confirmation)
> * Create blank workflow
> * Add a report
> * 'Next' to go to finalise page: Workflow Name is still read-only
> 
> The same thing happens if you Clone after Re-run.
> 
> This could go into a separate ticket, or be fixed in this branch if you prefer.

I have added changes and addressed all requests. However, I could not test certain functionality manually as my dev env is not rendering to test data for manual testing.